pipeline/download_gene.py

```diff
@@ -0,0 +1,154 @@
+    dataset_np : np.ndarray
+        Array containing gene expressions of the dataset.
+    expression_np : np.ndarray
+        Array containing image expression of the dataset.
+    metadata_dict : dict
+        Dictionary containing metadata of the dataset.
+        Keys are section numbers and image ids.
+    """
+    metadata_dict = {}
+    section_numbers = []
+    image_ids = []
+    expression_np = []
+    dataset_np = []
+
+    for img_id, section_coordinate, img, img_expression, df in dataset:
+        if section_coordinate is None:
+            # In `atldld.sync.download_dataset` if there is a problem during the download
+            # the generator returns `(img_id, None, None, None, None)
+            # TODO: maybe notify the user somehow?
+            continue
+
+        section_numbers.append(section_coordinate // 25)
+        image_ids.append(img_id)
+        warped_img = 255 - df.warp(img, border_mode="constant", c=img[0, 0, :].tolist())
+        warped_exp = df.warp(
+            img_expression, border_mode="constant", c=img[255, 255, :].tolist()
+        )
+        dataset_np.append(warped_img)
+        expression_np.append(warped_exp)
+
+    dataset_np = np.array(dataset_np)
+    expression_np = np.array(expression_np)
+
+    metadata_dict["section_numbers"] = section_numbers
+    metadata_dict["image_ids"] = image_ids
+    metadata_dict["image_shape"] = warped_img.shape
+
+    return dataset_np, expression_np, metadata_dict
+
+
+def main(
+    gene_name: str,
+    output_dir: Path | str | None = None,
+) -> int:
+    """Download gene expression dataset."""
+    # Imports
+    import json
+
+    from atldld.sync import DatasetDownloader
+    from atldld.utils import CommonQueries, get_experiment_list_from_gene
+
+    # To avoid Decompression Warning
+    PIL.Image.MAX_IMAGE_PIXELS = 200000000
+
+    # Download dataset on allen
+    if output_dir is None:
+        output_dir = Path(__file__).parent / "download-gene" / gene_name
+    else:
+        output_dir = Path(output_dir) / gene_name
+
+    if not output_dir.exists():
+        output_dir.mkdir(parents=True)
+
+    for axis in ["sagittal", "coronal"]:
+
+        experiment_list = get_experiment_list_from_gene(gene_name, axis)
+        for experiment_id in experiment_list:
+            dataset = DatasetDownloader(experiment_id, include_expression=True)
+            dataset.fetch_metadata()
+            dataset_gen = dataset.run()
+            axis = CommonQueries.get_axis(experiment_id)
+            dataset_np, expression_np, metadata_dict = postprocess_dataset(dataset_gen)
+            metadata_dict["axis"] = axis
+
+            np.save(output_dir / f"{experiment_id}-expression.npy", expression_np)
+            np.save(output_dir / f"{experiment_id}.npy", dataset_np)
+            with open(output_dir / f"{experiment_id}.json", "w") as f:
+                json.dump(metadata_dict, f)
+
+    return 0
+
+
+if __name__ == "__main__":
+    logging.basicConfig(level=logging.INFO)
```

Comment:
[NOT URGENT] this is changing the logging level of the `root` logger and it will be propagated
to all its children (including the `download-gene` logger). 

This can lead to unwanted verbosity from 3rd party package loggers. 

Anyway, we can easily improve this in the future.

Reply:
Perfect. Thanks!